notification_controller/controller/controller.go

```diff
@@ -61,7 +61,13 @@ func NewController(
 	secretName string,
 	configMapName string,
 ) *notificationController {
-	appClient := client.Resource(applications)
+	var appClient dynamic.ResourceInterface
+	namespaceableAppClient := client.Resource(applications)
+	appClient = namespaceableAppClient
+	if len(applicationNamespaces) == 0 {
+		appClient = namespaceableAppClient.Namespace(namespace)
+	}
+	appClient = client.Resource(applications)
```

Comment:
I think this line needs to be removed. Maybe we forgot to clean it up during the fix..